packages/client/src/execution/vmexecution.ts

```diff
@@ -179,21 +242,80 @@ export class VMExecution extends Execution {
       const td = await this.vm.blockchain.getTotalDifficulty(headBlock.header.hash())
       this.config.execCommon.setHardforkBy({ blockNumber: number, td, timestamp })
       this.hardfork = this.config.execCommon.hardfork()
-      this.config.logger.info(`Initializing VM execution hardfork=${this.hardfork}`)
-      if (number === BIGINT_0) {
-        const genesisState =
-          this.chain['_customGenesisState'] ?? getGenesis(Number(this.vm.common.chainId()))
-        if (!genesisState) {
-          throw new Error('genesisState not available')
+
+      if (this.config.execCommon.gteHardfork(Hardfork.Prague)) {
+        if (!this.config.statelessVerkle) {
+          throw Error(`Currently stateful verkle execution not supported`)
+        }
+        this.config.logger.info(`Skipping Vm verkle statemanager genesis hardfork=${this.hardfork}`)
```

Comment:
At least here in the logging: Vm -> VM. 🙂

Reply:
done